akka-actor/src/main/scala/akka/serialization/Serialization.scala

```diff
@@ -99,6 +100,25 @@ object Serialization {
         }
     }
   }
+
+  /**
+   * Use the specified @param system to determine transport information that will be used when serializing actorRefs
+   * in @param f code: if there is no external address available for the requested address then the systems default
+   * address will be used.
+   *
+   * @return value returned by @param f
+   */
+  def withTransportInformation[T](system: ExtendedActorSystem)(f: () ⇒ T): T = {
```

Comment:
we also need java api:
```
def withTransportInformation[T](system: ExtendedActorSystem, f: Supplier[T]): T = {
```

Reply:
Actually with Scala 2.12+ Java 8+ that overload would cause trouble. I think we need to rethink out strategy for adding those overloads (and in this case, not add one)

Reply:
I'm not sure to understand if/what I should do (something).

Reply:
let me try that

Reply:
I have verified that it works fine from Java when compiled with 2.12, and I think that is good enough for this feature, so nothing have to be done here.